File: client/src/views/models/tournament.ts
export interface Tournament {
	uid:string
	title:string
	statut:boolean
	description:string
	numberParticipate:number
	gameWay:string
	format:string
	deadlineDate:string
	winners:string[]
	tchat:string
	dateStart:string
	price:string[]
	priceParticipate:string
	rules:string
	isTeam:boolean
	maps:string
	laps:string[]
	server:string
	isPublic:boolean
	region:string
	game:Game
	plateform:Platform[]
	spectateur:string
}

export interface Game {
	uid:string
	name:string
	image:string
	logo:string
	slug:string
}

export interface Platform {
	uid:string
	description:string
	name:string
}

export interface User {
	uid:string
	firstname:string
	lastname:string
	email:string
	username:string
	avatar:string
	language:string
	point:number
}

File: server/models/tournament.go
package models

import "gorm.io/gorm"

/*type Tournaments struct {
	ID                uint
	Title             string  `gorm:"size:255;not null;"json:"title"`
	DateStart         string  `gorm:"size:255;not null;"json:"date_start"`
	Plateform         string  `gorm:"size:255;not null;"json:"plateform"`
	Price             float64 `gorm:"size:255;not null;"json:"price"`
	DeadlineDate      string  `gorm:"size:255;not null;"json:"deadline_date"`
	NumberParticipate int     `gorm:"size:255;not null;"json:"number_participate"`
	Statut            string  `gorm:"size:255;not null;"json:"statut"`
}*/

type Tournaments struct {
	gorm.Model
	Title             string
	DateStart         string
	Plateform         string
	Price             float64
	DeadlineDate      string
	NumberParticipate int
	Statut            string
}
